src/composeappmanager.cc

```diff
@@ -182,10 +192,18 @@ bool ComposeAppManager::fetchTarget(const Uptane::Target& target, Uptane::Fetche
   LOG_INFO << "Found " << cur_apps_to_fetch_and_update_.size() << " Apps to update";
 
   bool passed = true;
-  for (const auto& pair : cur_apps_to_fetch_and_update_) {
-    LOG_INFO << "Fetching " << pair.first << " -> " << pair.second;
-    if (!app_ctor_(pair.first).fetch(pair.second)) {
-      passed = false;
+  const auto& branch_to_pull = target.custom_data()["compose-apps-branch"].asString();
+  const auto& commit_hash_to_pull = target.custom_data()["compose-apps-hash"].asString();
+  if (!branch_to_pull.empty()) {
+    LOG_INFO << "Pulling from App Tree; branch: " << branch_to_pull << " commit: " << commit_hash_to_pull;
```

Comment:
as this develops, will you need to also assert that the device was configured to use ostree apps?

Reply:
I think about just doing fallback to the current way to pull apps. 
>the device was configured to use ostree apps?

Do we wanna another configuration variable? Or if an app's ostree repo exists/preloaded in a system image and Target has a commit to the app's ostree repo in Treehub then do the ostree-cased update otherwise fallback?

Reply:
I was hoping we could avoid the config variable and determine by the existence of the local repo

Reply:
ok, what entity should create the local repo, the preloading functionality, or the preloading and lmp-device-register if the repo is not preloaded?